src/pages/resultsView/mutualExclusivity/MutualExclusivityTab.tsx

```diff
@@ -0,0 +1,185 @@
+}
+
+export function calculatePValue(a: number, b: number, c: number, d: number): number {
+    return fisherTest(a, b, c, d).toNumber();
+}
+
+export function calculateLogOddsRatio(a: number, b: number, c: number, d: number): number {
+    return Math.log((a * d) / (b * c));
+}
+
+export function getMutuallyExclusiveCounts(data: MutualExclusivity[],
+    exclusive: (n: number) => boolean): [JSX.Element | null, JSX.Element | null] {
+
+    let exclusiveCount = null;
+    let significantCount = null;
+
+    const exclusiveData = data.filter(mutualExclusivity => exclusive(mutualExclusivity.logOddsRatio));
+    const significantData = exclusiveData.filter(mutualExclusivity => mutualExclusivity.pValue < 0.05);
+
+    const exclusiveLength = exclusiveData.length;
+    const significantLength = significantData.length;
+    if (exclusiveLength === 0) {
+        exclusiveCount = <span><b>no</b> gene pair</span>;
+    } else if (exclusiveLength === 1) {
+        exclusiveCount = <span><b>1</b> gene pair</span>;
+    } else {
+        exclusiveCount = <span><b>{exclusiveLength}</b> gene pairs</span>;
+    }
+
+    if (exclusiveLength > 0) {
+        if (significantLength === 0) {
+            significantCount = <span> (none significant)</span>;
+        } else {
+            significantCount = <span> (<b>{significantLength}</b> significant)</span>;
+        }
+    }
+
+    return [exclusiveCount, significantCount];
+}
+
+export function getCountsText(data: MutualExclusivity[]): JSX.Element {
+
+    const mutuallyExclusiveCounts = getMutuallyExclusiveCounts(data, n => n <= 0);
+    const coOccurentCounts = getMutuallyExclusiveCounts(data, n => n > 0);
+
+    return <p>The query contains {mutuallyExclusiveCounts[0]} with mutually exclusive alterations{
+        mutuallyExclusiveCounts[1]}, and {coOccurentCounts[0]} with co-occurrent alterations{
+        coOccurentCounts[1]}.</p>;
+}
+
+export function getData(isSampleAlteredMap: {[key: string]: boolean[]}): MutualExclusivity[] {
+
+    let data: MutualExclusivity[] = [];
+    const combinations = Combinatorics.combination(Object.keys(isSampleAlteredMap), 2).toArray();
+    combinations.forEach(combination => {
+
+        const geneA = combination[0];
+        const geneB = combination[1];
+        const counts = count(isSampleAlteredMap[geneA], isSampleAlteredMap[geneB]);
+        const pValue = calculatePValue(counts[0], counts[1], counts[2], counts[3]);
+        const logOddsRatio = calculateLogOddsRatio(counts[0], counts[1], counts[2], counts[3]);
+        const association = calculateAssociation(logOddsRatio);
+        data.push({geneA, geneB, pValue, logOddsRatio, association});
+    });
+    return data;
+}
+
+export function getFilteredData(data: MutualExclusivity[], mutualExclusivityFilter: boolean, coOccurenceFilter: boolean,
+                                significantPairsFilter: boolean): MutualExclusivity[] {
+
+    return data.filter(mutualExclusivity => {
```

Comment:
not neccessary to change but would this be more clearly written as
if (mutualExclusivityFilter && logOddsRatio <=0) return true;
...